game/osu/Beatmap.ksy

```diff
@@ -0,0 +1,146 @@
+meta:
+  id: beatmap
+  endian: le
+  imports:
+    - TimingPoint
+    - IntDoublePair
+    - String
+    - DateTime
+    - main
+seq:
+  - id: entry_size
+    type: u4
+  - id: artist_name
+    type: osu_string
+  - id: artist_name_unicode
+    type: osu_string
+  - id: song_title
+    type: osu_string
+  - id: song_title_unicode
+    type: osu_string
+  - id: creator
+    type: osu_string
+  - id: difficulty
+    type: osu_string
+  - id: audio_path
+    type: osu_string
+  - id: md5_hash
+    type: osu_string
+  - id: map_path
+    type: osu_string
+  - id: ranking
+    type: u1
+  - id: hitcircles
+    type: u2
+  - id: sliders
+    type: u2
+  - id: spinners
+    type: u2
+  - id: modtime
+    type: u8
+  - id: ar
+    type:
+      switch-on: _parent.osu_ver < 20140609

```

Comment:
could you verify that this compiles?

Reply:
this is a problem. I cannot run this with my ksv because of this error
```Compilation OK
... processing osu_file.ksy 0
Error: [{"file"=>"osu_file.ksy", "path"=>[], "message"=>"called attribute 'osu_ver' on generic struct expression 'Name(identifier(_parent))'"}]
Fatal errors encountered, cannot continue```

Reply:
you have 2 options.
1 merge everything into a single file
2. pass the variable as a `param`:
```yaml
params:
  - id: version
    type: u8
```